### ISSUE TITLE
Questions

### ISSUE THREAD

**Issue by Sandrillka** [sukuidoardo/PYtibiaAutoHeal#1]
Hey, just have a couple of questions. I'm looking for some heal bot for 13+ tibia OT server.
1. Do I need OBS for it to run?
2. Do I need to download or install anything besides that?
Hope to get some answers cheers.

**Comment by sukuidoardo** [comment]
Hey there, it doesnt need any external program to run but you may need to install python and included libraries in the code. You can get it from python's official website for free. Libraries can be installed in cmd or terminal with a command "pip install (library name)".
Honestly I didnt expect anyone to find this. I am not really a coder so I cant really help with the process of installing but if you are familiar with coding or python in general it should be easy. 
I may add a standalone version later if you are still interested but I'm kind of in the middle of writing my thesis so no promises for upcoming 2-3 months.
Just make sure the OT you are playing is ok with some form of heal botting/mild automatisation. And make sure to test it in a safe enviroment first. If you have problems running it you can ask me questions. But instructions I put up should be enough for most things.

**Comment by sukuidoardo** [comment]
Also let me know if you successfully used the bot :D

**Comment by Sandrillka** [comment]
Hi, so I installed python but when I open your script it instantly closes. I'm really green in coding. Could you tell me which libraries should I install for it to work?

**Comment by sukuidoardo** [comment]
pip install pillow pyautogui pynput

running this on cmd should get you all the libraries in one go, if not try them one by one like "pip install pillow", "pip install pyautogui" etc

tkinter should be included with python itself but if you will still get errors try installing it too "pip install tkinter". Hope it works for you, let me know.
If it still doesnt work I can continue to help and later on I'll try to make a standalone version without needing to install anything. Just need to have some procrastination time :D

**Comment by Sandrillka** [comment]
When I open python and type " pip install pillow " or any other i get "SyntaxError: Invalid syntax" message.
 In the python folder libs I found  "_tkinter.lib" so yeah it installed with python installation like you said. But can't seem to download anything. So if you get any advice it would be cool. Like I said I'm totally green to this.

**Comment by sukuidoardo** [comment]
You need to open cmd as in command prompt in windows not python. It opens a black window, then you can type above instructions on there. If you have the latest windows you can just type "cmd" in the search bar at the bottom of your desktop. Or you can find "run" and type cmd in there. I'll add an image here about it.

Once you install all the libraries try running the program again and let me know
![images-19](https://github.com/user-attachments/assets/e3336d6a-68f1-4395-a180-10175229f22d)

**Comment by Sandrillka** [comment]
I tried it aswell since it was the first cmd that came to my mind but there i had this message:
'pip' is not recognized as an internal or external command,
operable program or batch file.

**Comment by sukuidoardo** [comment]
Huh it should have worked if python is installed. In this case I cant really help you man, like I said I'll try to make a standalone version but I dont really have free time to do so for a while. If you are tracking this thread with e mail I can update here when I make it.

**Comment by Sandrillka** [comment]
Yeah I understand and thank you for trying to help me. I'll take a look from time to time and good luck with your studies.
@edit
So i fixed the problem. Watched some tutorial about this error I had ( Download path was wrong in system environment )
downloaded the libraries you mentioned and the script worked. I'll take a look soon and give you info about how it works.
@edit2
The script uses exura for example but it spams it constantly no matter my HP even when I set a pixel at which it should go off.
Could you show me a SS how it looks in your client for example and maybe how you set it up if that's not a problem? thanks!

**Comment by sukuidoardo** [comment]
It's some progress atleast. Which OTS are you playing on if you tell me I can test on their client to see if it works and let you know. 

In the meantime try to select the pixel on your hp bar on the side panels. If it doesnt work stop monitoring and re select the pixel. Maybe make a fresh character and try it on that so if you die its no big deal.

**Comment by Sandrillka** [comment]
I restarted it few times and it worked. Thank you very much for help and just going to ask a few questions to be sure.
For mage I should set my healing spells as:
exura               <-- priority 3
exura gran       <-- priority 2
exura vita         <-- priority 1
So the vita has highest priority in case im low health and it wont use exura correct?

and for knight I should also 
check "always check" for my exura ico and health potions boxes so i'll heal from both if im under heavy attack correct?

**Comment by sukuidoardo** [comment]
https://youtu.be/LK2d6DgVWtM?si=7Uuj_x93pPXbHuWk

Couldnt get stupid screen recording working so here is a video recorded on my phone about how it works for me. Though you already figured by the time :D

As for your questions, both are correct, be mindful of your mana potion spam on knight though it may override heal potions

**Comment by Sandrillka** [comment]
Yeah had the same thought about it spamming mana/heal potions on knights in a mob trap with mana drain mobs around.(Don't think the higher priority on HP would work over MP since it would be for different resources right?)
Anyway thank you very much for all the help and time! If I have any more questions I'll let you know. Have a nice day.

**Comment by sukuidoardo** [comment]
I think in theory you can set up another pixel on mana bar and set it to lower priority then set a heal potion one with higher priority so it will override mana potion check when your hp gets low. Set exura ico to always check and it should work.

For mages since you dont heal with potions you can set up an always check mana bar pixel and use different heals like you mentioned before.

Priorities work on pixel color changes only, so doesn't matter if its on mana bar or hp bar. If you set a pixel on mana bar with lower prio and set another pixel on hp bar with highee prio once it detects a change in color on hp bar it should just ignore mana bar changes till you get hp back.

Though I would still use mana potions manually because when in danger you often need to spam hp potions and you dont want a mana potion to be used inbetween accidentally

**Comment by shokh4n** [comment]
Does it work in Global with OBS?

**Comment by sukuidoardo** [comment]
> Does it work in Global with OBS?

You should be able to run it without any external program. You might need to install some python libraries depending on stuff but I have used it for many ots, it should work as intended. Mind that it WILL try to press whatever hotkey you assigned to a pixel if you alt tab or something, so make sure to pause the program before alt tabing to something else

If by global you mean official tibia servers. I'd avoid using this script, it will most likely get you banned
